src/Products/CMFPlone/browser/exceptions.py

```diff
@@ -30,7 +30,15 @@ def __call__(self):
             # and sets the proper location header
             return
 
-        self.context = self.__parent__
+        # Always use portal root as context to ensure resources load correctly
+        current = self.__parent__
+        while hasattr(current, "__parent__") and current.__parent__ is not None:
+            if hasattr(current, "meta_type") and current.meta_type == "Plone Site":
+                break
+            current = current.__parent__
+
+        self.context = current
+        self.__parent__ = current  # Also update __parent__ to ensure consistency
         request = self.request
```

Comment:
This might cause issues on some set up.

If I understand it correctly, you want to get the **navigation root**.

I would check if it works with:
```
from plone.base.navigationroot import get_navigation_root_object
```

In addition, this might break the suggestion:
https://github.com/plone/Products.CMFPlone/blob/2122f8bfe0cb46b8d21f652f3f2c0f0e24856274/src/Products/CMFPlone/browser/templates/error_message.pt#L43-L44

Most probably the issue lies elsewhere.
Gut feeling: I would check the styles viewlet.